enhancements/clusterGroupUpgradeStatus.md

```diff
@@ -0,0 +1,319 @@
+  - **timeout**: the policy applied to enforce remediation but it does not become compliant during the timeout limits defined in the remediationStrategy.
+
+The cluster state has 2 possible state;
+  - **complete**: if all the policies has a compliant state on the cluster
+  - **inProgress**: if all the policies are in compliant/nonCompliant or notApplied state.
+  - **failed**: if at least 1 policy has timeout state.
+
+
+```
+apiVersion: cluster.open-cluster-management-extension.io/v1beta1
+kind: ClusterGroupUpgrade
+metadata:
+  name: cgu-upgrade-new
+  namespace: default
+spec:
+  enable: true
+  clusters:
+  - spoke2
+  - spoke3
+  - spoke4
+  - spoke5
+  managedPolicies:
+    - policy1-common
+    - policy2-group
+    - policy3-site
+  remediationStrategy:
+    canaries:
+    - prod1
+    maxConcurrency: 2
+    timeout: 240
+status:
+ conditions: 
+   ...
+ canaryClusters:
+ - name: spoke1
+   policies:
+     policy1-common: compliant
+     policy2-group: compliant
+     policy3-site:  compliant
+   state: complete
+ clusters:
+ - name: spoke2
+   policies:
+     policy1-common: compliant
+     policy2-group: nonCompliant
+     policy3-site:  notApplied
+   state: inProgress
+ - name: spoke3
+   policies:
+     policy1-common: compliant
+     policy2-group: timeout
+     policy3-site:  notApplied
+   state: failed
+ - name: spoke4
+   policies:
+     policy1-common: notApplied
+     policy2-group: notApplied
+     policy3-site:  notApplied
+   state: inProgress
+ - name: spoke5
+   policies:
+     policy1-common: notApplied
+     policy2-group: notApplied
+     policy3-site:  notApplied
+   state: inProgress
+```
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### CGU status 
+
+##### 1- copied policies, placement rules and placement binding names
+
+The CGU controller copies the ACM policies defined under the CGU->managedPolicies field then change its remediation to enforce and create a placement rule and placement binding accordingly to the select clusters.
+While storing the copied policies, placement rules and placement binding names in the CGU status is required by the current implementation of the reconcile process however its not mandatory. Having the copied policies names consist of the CGU name and policy index will make it unique for the reconciling loop.
+Moreover, based on the [selective policy enhamcement](https://github.com/open-cluster-management-io/enhancements/pull/57) there is no needs to copies policies for future integration with ACM coming releases.  
+
+##### 2- batches
+
+The CGU status field store the current cluster batch number in order to iterate to the next batch after success/failed of the running batch.  The new proposed CGU status does not require that as all clusters with indies are stored under clusters list field.
+The iteration of the batches can be determined by the maxConcurrency number defined under the remediationStrategy and the selected clusters list.
```

Comment:
I am confused by this. I can't figure out how we determine the current batch number? Can we have more explanation on what is the selected clusters list?

Reply:
the idea is for 10 clusters and maxConcurrency equal to 4, the first 4 clusters (1-4) in the cluster list will be the first batch then the next 4 cluster (5-8) will be the next batch and so on. Will clarify it more

Reply:
it based on the cluster state if it is inProgress then still active when it is failed/timeout we move to the next one